File: STRIVER/video/vid-5/pattern9/p9.py
    #      *
    #     * *     
    #    * * *   
    #   * * * *  
    #  * * * * *  
    #   * * * *  
    #    * * *   
    #     * *     
    #      *


def p9(row:int)->None:
    # upper
    for i in range(row):
        # space
        for j in range(row-i-1):
            print("  ",end="")

        # stars
        for j in range((2*i+1)):
            print("* ",end="")

        # space
        for j in range(row-i-1):
            print("  ",end="")

        print("")

    #lower 
    for i in range(row+1):
        # space
        for j in range(i):
            print("  ",end="")

        # stars
        for j in range(2*row - (2*i+1)):
            print("* ",end="")

        # space
        for j in range(i):
            print("  ",end="")

        print("")

row = 3
p9(row)

File: STRIVER/video/vid-5 (Patterns)/pattern9/p9.cpp
//      *
//     * *
//    * * *
//   * * * *
//  * * * * *
//  * * * * *
//   * * * *
//    * * *
//     * *
//      *

#include <iostream>
using namespace std;

int main()
{
    int row = 3;

    // upper
    for (int i = 0; i < row; i++)
    {
        // space
        for (int j = 0; j < row-i-1; j++)
        {
            cout << "  ";
        }
        // starts
        for (int j = 0; j < (2 * i + 1); j++)
        {
            cout << "* ";
        }
        // space
        for (int j = 0; j < row-i-1; j++)
        {
            cout << "  ";
        }
        cout << endl;
    }
    // lower
    for (int i = 0; i < row; i++)
    {
        // space
        for (int j = 0; j < i; j++)
        {
            cout << "  ";
        }
        // starts
        for (int j = 0; j < 2 * row - (2 * i + 1); j++)
        {
            cout << "* ";
        }
        // space
        for (int j = 0; j < i; j++)
        {
            cout << "  ";
        }
        cout << endl;
    }
    return 0;
}